### PR TITLE
Xcode 10, Swift 4.2, Stencil 0.13.0

### PR DESCRIPTION
**Pull request by djbe**
This updates the project to use Xcode 10 and Swift 4.2.

First order of business is a Stencil issue, where iteration through a Dictionary is now random due to the Hashable changes.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ SCHEME_NAME = 'swiftgen'.freeze
 CONFIGURATION = 'Debug'.freeze
 RELEASE_CONFIGURATION = 'Release'.freeze
 POD_NAME = 'SwiftGen'.freeze
-MIN_XCODE_VERSION = 9.2
+MIN_XCODE_VERSION = 10.0
```
Comment (AliSoftware) [inline Rakefile]: Nice I think I'd have forgotten that one 👏

**Comment by djbe** [comment]
Oh I wouldn't merge this yet, this is just to keep track of things. First fix stuff in Stencil, then we can test things here to see if everything is 👌 .

**Comment by djbe** [comment]
Related Stencil PR: https://github.com/stencilproject/Stencil/pull/240

**Comment by AliSoftware** [comment]
Needs a rebase?

**Comment by djbe** [comment]
No, why? There are some bigger context/template changes needed to fix the tests with xcode 10, they're in a separate PR: #504.

**Comment by AliSoftware** [comment]
I just assumed due to the build failure and since we now merged the stencil update?

**Comment by djbe** [comment]
Nah there are some underlying issues preventing correct testing, see #504.

**Pull request merged**